When a submake fails, the build of FLTK was stopping, but the main
make was returning 0. Therefore, Buildroot was going on with the build
of the other components, even though the build of FLTK failed.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: fltk-1.1.7/Makefile
===================================================================
--- fltk-1.1.7.orig/Makefile	2010-05-21 15:52:20.000000000 +0200
+++ fltk-1.1.7/Makefile	2010-05-21 15:52:43.000000000 +0200
@@ -32,7 +32,7 @@
 all: makeinclude
 	for dir in $(DIRS); do\
 		echo "=== making $$dir ===";\
-		(cd $$dir; $(MAKE) $(MFLAGS)) || break;\
+		(cd $$dir; $(MAKE) $(MFLAGS)) || exit 1;\
 	done
 
 install: makeinclude
@@ -42,7 +42,7 @@
 	-chmod 755 $(DESTDIR)$(bindir)/fltk-config
 	for dir in FL $(DIRS); do\
 		echo "=== installing $$dir ===";\
-		(cd $$dir; $(MAKE) $(MFLAGS) install) || break;\
+		(cd $$dir; $(MAKE) $(MFLAGS) install) || exit 1;\
 	done
 
 install-desktop: makeinclude
@@ -54,7 +54,7 @@
 	$(RM) $(DESTDIR)$(bindir)/fltk-config
 	for dir in FL $(DIRS); do\
 		echo "=== uninstalling $$dir ===";\
-		(cd $$dir; $(MAKE) $(MFLAGS) uninstall) || break;\
+		(cd $$dir; $(MAKE) $(MFLAGS) uninstall) || exit 1;\
 	done
 
 uninstall-desktop: makeinclude
@@ -65,14 +65,14 @@
 depend: makeinclude
 	for dir in $(DIRS); do\
 		echo "=== making dependencies in $$dir ===";\
-		(cd $$dir; $(MAKE) $(MFLAGS) depend) || break;\
+		(cd $$dir; $(MAKE) $(MFLAGS) depend) || exit 1;\
 	done
 
 clean:
 	-$(RM) core *.o
 	for dir in $(DIRS); do\
 		echo "=== cleaning $$dir ===";\
-		(cd $$dir; $(MAKE) $(MFLAGS) clean) || break;\
+		(cd $$dir; $(MAKE) $(MFLAGS) clean) || exit 1;\
 	done
 
 distclean: clean
